### PR TITLE
Setup force push check

### PR DESCRIPTION
**Pull request by jameesjohn**
## Explanation
Checks that users do not force push to a pull request.
This has been tested at https://github.com/jameesjohn/comment-on-pr/pull/36

## Checklist
- [x] I have successfully deployed my own instance of Oppiabot.
  - You can find instructions for doing this [here](https://github.com/oppia/oppiabot/wiki/Deploying-your-own-instance-of-the-oppiabot).
- [x] I have manually tested all the changes made in this PR following the [manual tests matrix](https://github.com/oppia/oppiabot/wiki/Manual-Tests-Matrix).

### CODE REVIEW

File: constants.js [hunk 1]
```diff
@@ -38,15 +40,16 @@ const checksWhitelist = {
...
+    [pushEvent]: [forcePushCheck]
   },
   'oppiabot': {
     [openEvent]: [claCheck],
     [reopenEvent]: [],
     [synchronizeEvent]: [mergeConflictCheck],
     [closeEvent]: [allMergeConflictCheck],
     [editEvent]: []
-  }
+  },
```
Comment (ankita240796) [inline constants.js:52]: Remove the comma.

File: constants.js [hunk 2]
```diff
@@ -55,8 +58,9 @@ module.exports.labelEvent = labelEvent;
 module.exports.synchronizeEvent = synchronizeEvent;
 module.exports.closeEvent = closeEvent;
 module.exports.editEvent = editEvent;
-module.exports.issuesLabelEvent = issuesLabelEvent
-module.exports.issuesAssignedEvent = issuesAssignedEvent
+module.exports.issuesLabelEvent = issuesLabelEvent;
```
Comment (ankita240796) [inline constants.js:61]: Let us rename labelEvent to PRLabelEvent to avoid confusion.

File: index.js [hunk 3]
```diff
@@ -134,4 +138,12 @@ module.exports = (oppiabot) => {
       await runChecks(context, constants.editEvent);
     }
   });
+
+  oppiabot.on('push', async (context) => {
```
Comment (ankita240796) [inline index.js:142]: We want this check only for pull requests not for branches directly pushed any time like develop/release branches. Can we skip for that?

File: lib/checkBranchPush.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    const user = context.payload.sender.login;
+    const link = 'here'.link(
+      'https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia' +
+      '#instructions-for-making-a-code-change');
+    const commentParams = context.repo({
+      body: 'Hi @'+ user +
+      ', force pushing is not allowed as it makes code reviews hard. ' +
+      'You can learn more about this '+ link + '. I’ll be closing this, ' +
+      'please make a new PR with the required changes. Thanks!',
```
Comment (ankita240796) [inline lib/checkBranchPush.js:53]: Mention the specific point which talks about avoiding force push.

File: lib/checkBranchPush.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      // Wait for 2 seconds before searching so that the pull request
+      // would have been synchronized with the new commit.
+      await sleep(2000);
+      const pullRequestData = await context.github.search.issuesAndPullRequests({
+        q: `${sha} repo:${context.payload.repository.full_name}`,
+      });
+      // Since we are searching via the sha, only one PR will be found,
+      // which is the PR that we are looking for.
+      pullRequest = pullRequestData.data.items[0]
+    } while(pullRequest === undefined);
```
Comment (ankita240796) [inline lib/checkBranchPush.js:43]: Should we do something to ensure that the check isn't stuck in the while loop for more than x seconds?

Comment (jameesjohn) [reply lib/checkBranchPush.js]: Done

File: lib/checkBranchPush.js (hunk 4 above)
Comment (jameesjohn) [reply lib/checkBranchPush.js:70]: We can't get a direct link to that point.

File: constants.js [hunk 6]
```diff
@@ -33,12 +35,13 @@ const checksWhitelist = {
...
+    [PRLabelEvent]: [assigneeCheck],
     [synchronizeEvent]: [mergeConflictCheck, jobCheck],
     [closeEvent]: [allMergeConflictCheck],
     [editEvent]: [wipCheck],
     [issuesLabelEvent]: [issuesLabelCheck],
-    [issuesAssignedEvent]: [issuesAssignedCheck]
+    [issuesAssignedEvent]: [issuesAssignedCheck],
+    [pushEvent]: [forcePushCheck]
   },
   'oppiabot': {
```
Comment (ankita240796) [inline constants.js:46]: Let us add all the events here as well with empty lists.

File: lib/checkBranchPush.js (hunk 4 above)
Comment (ankita240796) [reply lib/checkBranchPush.js:70]: I meant just add that it is point x specifically.

File: constants.js (hunk 6 above)
Comment (jameesjohn) [reply constants.js:46]: Done.

File: lib/checkBranchPush.js (hunk 4 above)
Comment (jameesjohn) [reply lib/checkBranchPush.js:70]: Done.

**Pull request merged**